src/main/java/executor/service/model/request/ScenarioRequest.java

```diff
@@ -0,0 +1,31 @@
+package executor.service.model.request;
+
+import executor.service.model.Scenario;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
+
+import java.util.List;
+
+/**
+ * Represents a request containing a list of scenarios and a user ID.
+ * This class is used for input validation and data transfer.
+ *
+ * @author Oleksii Bondarenko
+ * @version 01
+ * @see Scenario
+ */
+public record ScenarioRequest(
+
+        @Valid
```

Comment:
@NotEmpty is better

Reply:
1. Using @Valid:
@Valid specifies that the scenarios field must pass the validation defined in the Scenario class (and its sub-objects, if any).
This option is useful if you need to ensure that each element in the list of scenarios is also validated according to the rules defined in the Scenario class.
This is suitable if you want to ensure that all scenarios in the list comply with your business rules.

2. Using @NotEmpty:
@NotEmpty ensures that the scenarios field is not empty (that is, the list is not null or empty).
This option is useful if you simply care that the list of scripts is provided and not empty.
However, it does not validate the contents of the list, so the internal details of each script may not meet your requirements.

Conclusion: for our tasks it is necessary to use @Valid.